docs/design/pnc.md

```diff
@@ -0,0 +1,113 @@
+Projects that currently use OSBS to fetch sources from PNC do so through a file named `fetch-artifacts-pnc.yaml`
+([docs](https://osbs.readthedocs.io/en/osbs_ocp3/users.html?highlight=fetch#fetch-artifacts-pnc-yaml)).
+Example of a `fetch-artifacts-pnc.yaml` file:
+
+```yaml
+metadata:
+  # this object allows additional parameters, you can put any metadata here
+  author: shadowman
+builds:
+  # all artifacts are grouped by builds to keep track of their sources
+  - build_id: '1234' # build id must be string
+    artifacts:
+      # list of artifacts to fetch, artifacts are fetched from PNC using their IDs
+      - id : '12345' # artifact id must be string
+        # the target can just be a filename or path+filename
+        target: test/rhba-common-7.10.0.redhat-00004.pom
+      - id: '12346'
+        target: prod/rhba-common-7.10.0.redhat-00004-dist.zip
+  - build_id: '1235'
+    artifacts:
+      - id: '12354'
+        target: test/client-patcher-7.10.0.redhat-00004.jar
+      - id: '12355'
+        target: prod/rhdm-7.10.0.redhat-00004-update.zip
+```
+
+These artifacts are fetched to `artifacts/<target>` path at the root of the repository.
+
+### PNC instance configuration
+Currently, the url of the PNC instance is configured as a part of OSBS config. 
+
+## Overview of the current implementation in OSBS
+
+[atomic_reactor/plugins/fetch_maven_artifacts.py](https://github.com/containerbuildsystem/atomic-reactor/blob/master/atomic_reactor/plugins/fetch_maven_artifacts.py)
+1) OSBS loads the `fetch-artifacts-pnc.yaml`
+2) Information about individual artifacts is fetched from PNC api at `/artifacts/{id}` endpoint
+   - `publicUrl` key is used to get download url
+   - supported checksums are saved
+   - build ids are saved
+3) Download queue is formed containing above info for each artifact
+4) SBOM component info for each artifact is fetched from PNC @ `/artifacts/{id}`, `purl` key
+5) Queue is executed and each artifact is downloaded, checksum verified
+6) SBOM is compiled, including PNC components from step 4)
+
+
+### Missing features
+
+PNC SBOM components don't have the most complete purl, e.g.:
+`pkg:maven/org.kie.trustyai/explainability-service@999.0.0.managedsvc-redhat-00001?type=zip`
+
+According to [purl spec](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#maven), `classfier` and
+`type` parameters should be also defined, method is currently pending depending on if the artifacts are single components,
+or products that contain other components. Purl should also specify a `repository_url` parameter.
+
+
+
+## Design for the implementation in Cachi2
+
+
+### Providing the content for the hermetic build
+
+#### PNC config
+Existing `fetch-artifacts-pnc.yaml` file ([docs](https://osbs.readthedocs.io/en/osbs_ocp3/users.html?highlight=fetch#fetch-artifacts-pnc-yaml))
+will be used as configuration. It needs to be extended to also include PNC instance url.
+
+#### Pre-fetch PNC dependencies
```

Comment:
I feel like this section should be standalone and separate from "Providing content for the hermetic build". This is because, during the build, the prefetch has already happened. Those are two distinct phases: prefetch -> build.

(and remember that we consider that the `buildah bud`/`podman build` command is executed with the `--network none` flag, so there's absolutely no network access).

Reply:
Moved